src/rebar_erlc_compiler.erl

```diff
@@ -81,111 +89,189 @@
 %%                           'old_inets'}]}.
 %%
 
--spec compile(rebar_app_info:t()) -> 'ok'.
-compile(AppInfo) ->
+%% @equiv compile(AppInfo, []).
+
+-spec compile(rebar_app_info:t()) -> ok.
+compile(AppInfo) when is_tuple(AppInfo), element(1, AppInfo) == app_info_t ->
```

Comment:
can get rid of the whole guard with `is_record(AppInfo, app_info_t)`, or at the very least just drop `is_tuple/1` given `element/2` encompasses it.